lib/plugins/sfDoctrinePlugin/lib/task/sfDoctrineBuildTask.class.php

```diff
@@ -185,61 +187,63 @@ protected function execute($arguments = [], $options = [])
             }
         }
 
-        if (self::BUILD_SQL == (self::BUILD_SQL & $mode)) {
-            $task = new sfDoctrineBuildSqlTask($this->dispatcher, $this->formatter);
-            $task->setCommandApplication($this->commandApplication);
-            $task->setConfiguration($this->configuration);
-            $ret = $task->run();
+        if (!$is_production) {
+            if (self::BUILD_SQL == (self::BUILD_SQL & $mode)) {
+                $task = new sfDoctrineBuildSqlTask($this->dispatcher, $this->formatter);
+                $task->setCommandApplication($this->commandApplication);
+                $task->setConfiguration($this->configuration);
+                $ret = $task->run();
 
-            if ($ret) {
-                return $ret;
+                if ($ret) {
+                    return $ret;
+                }
             }
-        }
 
-        if ($options['and-migrate']) {
-            $task = new sfDoctrineMigrateTask($this->dispatcher, $this->formatter);
-            $task->setCommandApplication($this->commandApplication);
-            $task->setConfiguration($this->configuration);
-            $ret = $task->run();
+            if ($options['and-migrate'] ?? false) {
```

Comment:
There are some default values for tests like this, too, in order to avoid warnings about unset array elements.